### ISSUE TITLE
Common dependencies for Bioinformatics

### ISSUE THREAD

**Issue by verdurin** [hpcugent/easybuild-easyconfigs#4116]
An attempt to resurrect the previous discussion in #1689 about common dependencies for Bioinformatics easyconfigs.

@fgeorgatos 
@pescobar 
@jerowe 
@rjeschmi 

```
dependencies = [
    ('bzip2', '1.0.6'),
    ('zlib', '1.2.8'),
    ('libreadline', '6.3'),
    ('ncurses', '5.9'),
# **the ones beyond this line are also known as biodeps**
    ('Boost', '1.58.0', '-Python-2.7.10'),
    ('SAMtools', '1.3.1'),
    ('Perl', '5.20.0', '-bare'),
    ('Java', '1.7.0_80', '', True),
    ('libpng', '1.6.17'),
]
```

**Comment by pescobar** [comment]
I would be happy if we finally manage to agree on common deps for bioinfo tools ;)

I would suggest to use `libreadline/6.2` instead of `libreadline/6.3`. This way we can avoid the need of this patch https://github.com/hpcugent/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/l/libreadline/libreadline-6.3-bugfix.patch

I use version 6.2 as the default in my cluster and it works fine for me so I don't see the need to upgrade as 6.2 is well tested and working and libreadline is just an "auxiliary" lib. Version 6.2 is also the one available in the centos7 repos. Or maybe someone has a technical reason which justifies upgrading to 6.3?

I also agree to stay with `ncurses/5.9` for the same reason. it's well tested and working and it's the available version in centos7

**Comment by fgeorgatos** [comment]
agreed with both of you; also, try to settle on the following points,
they are very contentious:

```

    ('Boost', '1.58.0', '-Python-2.7.10'),
    ('SAMtools', '1.3.1'),

```

Once that's done, it would be great if we/you could coordinate also on
a common toolchain,
at least for the PR contributions part. Suggestion: let's settle on
this after a biweekly call.

**Comment by rjeschmi** [comment]
I'm happy to follow along with what others are doing in general. I haven't had too much trouble with newer deps, but need to keep around older versions of things since users are fussy.
